File: app/src/main/java/com/popalay/cardme/presentation/screens/holders/HoldersFragment.java
package com.popalay.cardme.presentation.screens.holders;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.popalay.cardme.R;
import com.popalay.cardme.data.models.Holder;
import com.popalay.cardme.databinding.FragmentHoldersBinding;
import com.popalay.cardme.presentation.base.BaseFragment;
import com.popalay.cardme.presentation.base.ItemClickListener;
import com.popalay.cardme.presentation.screens.holderdetails.HolderDetailsActivity;

import java.util.List;

public class HoldersFragment extends BaseFragment implements HoldersView, ItemClickListener<Holder> {

    @InjectPresenter HoldersPresenter presenter;

    private FragmentHoldersBinding b;

    private HoldersViewModel viewModel;

    public static HoldersFragment newInstance() {
        return new HoldersFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        b = DataBindingUtil.inflate(inflater, R.layout.fragment_holders, container, false);
        return b.getRoot();
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initUI();
    }

    @Override
    public void setHolders(List<Holder> holders) {
        viewModel.setHolders(holders);
    }

    @Override
    public void openHolderDetails() {
        startActivity(HolderDetailsActivity.Companion.getIntent(getContext()));
    }

    @Override
    public void onItemClick(Holder item) {
        presenter.onHolderClick(item);
    }

    private void initUI() {
        viewModel = new HoldersViewModel();
        b.setModel(viewModel);
        b.setListener(this);
    }
}

File: app/src/main/java/com/popalay/cardme/presentation/screens/holders/HoldersFragment.kt
package com.popalay.cardme.presentation.screens.holders

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.popalay.cardme.R
import com.popalay.cardme.databinding.FragmentHoldersBinding
import com.popalay.cardme.presentation.base.BaseFragment
import javax.inject.Inject

class HoldersFragment : BaseFragment() {

    @Inject lateinit var factory: ViewModelProvider.Factory

    companion object {
        fun newInstance() = HoldersFragment()
    }

    private lateinit var b: FragmentHoldersBinding

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        b = DataBindingUtil.inflate<FragmentHoldersBinding>(inflater, R.layout.fragment_holders, container, false)
        ViewModelProviders.of(this, factory).get(HoldersViewModel::class.java).let {
            b.vm = it
        }
        return b.root
    }
}
